weatherman.py

```diff
@@ -0,0 +1,158 @@
+import os
+import csv
+import argparse
+from colorama import Fore, Style
+
+def convert_month(month_input):
+    month_names = [
+        "Jan", "Feb", "Mar", "Apr", "May", "Jun",
+        "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
+    ]
+    if month_input.isdigit() and 1 <= int(month_input) <= 12:
+        return month_names[int(month_input) - 1]
+    return None
+
+def read_files(year, month):
+    headers = []
+
+    for iterate in os.listdir(r"D:\Cogent Labs\github\weatherman\weatherfiles"):
+        if year in iterate and (not month or month in iterate):
+            file_path = os.path.join(r"D:\Cogent Labs\github\weatherman\weatherfiles", iterate)
+
+            with open(file_path, 'r') as csvfile:
+                reader = csv.DictReader(csvfile)
+                for row in reader:
+                    headers.append(row)
+    
+    return headers
+
+def calculate_statistics(headers):
+    highest_temp = float('-inf')
+    lowest_temp = float('inf')
+    most_humid_day = None
+    most_humidity = 0
+    meandata = 0
+    mean_humidity = 0
+    average_lowest_temp = 0
+    average_highest_temp = 0
+
+    for row in headers:
+        max_temp = row['Max TemperatureC']
+        min_temp = row['Min TemperatureC']
+        humidity = row['Max Humidity']
+        m_humidity = row[' Mean Humidity']
+
+        try:
+            if max_temp and float(max_temp) > highest_temp:
```

Comment:
Perform calculations based on the input provided, rather than executing all calculations for every inpu